torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -510,13 +510,36 @@ struct PythonPrintImpl {
     }
     indent();
     printValueList(body_, lhs);
+    // We need to preserve Union/Optional type annotations, but only if
+    // we're not assigning values as part of a tuple unpacking statement
```

Comment:
Does this mean unpacking tuple into a `union`-typed variable is not supported now? If so, let's add an explicit error and a `TODO` for future fix.

Reply:
Can you give an example? We only add annotations on the first assignment of a variable, and I can't think of a case in which some lhs could be a `Union` without an annotation. (Obviously, having two Union annotations on the same line is a SyntaxError!) Thank you!

Reply:
Thanks for clarification, I didn't know about this Python limitation before.